File: pascal/ContagemSuperinteligente.pas
program 'Contagem SuperInteligente';
Var
    count, n: Integer;
begin
    repeat
    writeln('------------MENU------------');
    writeln('|[1] Para contar de 1 a 10. |');
    writeln('|[2] Para contar de 10 a 1. |');
    writeln('|[3] Para sair.             |');
    read(n);
    case n of
        1: begin
            count:= 0;
            repeat
                count:= (count + 1);
                writeln(count, '...');
            until (count >= 10);
        end;
        2: begin
            count:= 10;
            repeat
                writeln(count, '...');
                count:= (count - 1);
            until (count <= 0);
        end;
        3: begin
        end;
    end;
    until (n = 3);
end.

File: python/contagemsuperinteligente.py
c = 0
while (c < 3):
  print('------------------------------')
  print('             MENU             ')
  print('------------------------------')
  print('|[1] Para contar de 1 a 10.   |')
  print('|[2] Para contar de 10 a 1.   |')
  print('|[3] Para sair.               |')
  c = int(input())
  match c:
    case 1:
      count = 1
      while (count <= 10):
        print(count, '...')
        count = (count + 1)
    case 2:
      count = 10
      while (count >= 1):
        print(count, '...')
        count = (count - 1)
    case 3:
      print('Saindo...')